apps/mobile/src/components/OfferDetailScreen/api.ts

```diff
@@ -79,19 +83,35 @@ export function useReportOfferHandleUI(): (
             offerId,
           })
         }),
+        TE.mapLeft((e) => {
+          if (e._tag !== 'UserDeclinedError') {
+            Alert.alert(toCommonErrorMessage(e, t) ?? t('common.unknownError'))
+          }
+          loadingOverlay.hide()
+          return false
+        }),
+        TE.map(() => {
+          store.set(reportedFlagAtom, true)
+        }),
+        TE.chainFirstW(() => {
```

Comment:
It might be a good practice to change this to `TE.chainFirstTaskK`, it does not matter if TE results in Left value, right? With this code, it would not hide loadingOverlay, also it would not go back, since in the following match, it would trigger onLeft.

Reply:
Left value could be retuned if user presses back button for example.

Reply:
As there will not be any "negative" button present in the dialog it should not happen that on left is triggered, but this is definitely cleaner solution

Reply:
But I get it that in the future the layout can change, so this way we will not have any problems